jumpscale/core/base/factory.py

```diff
@@ -514,11 +514,24 @@ def find(self, name):
         """
         instance = super().find(name)
         if not instance:
-            try:
-                instance_config = self.store.get(name)
-            except FileNotFoundError:
-                return None
-            if not instance_config:
-                return None
-            instance = self.type.from_dict(instance_config)
+            instance = self.load_from_store(name)
+        return instance
+
+    def load_from_store(self, name):
+        """loads instance from store
+
+        Args:
+            name (str): instance name
+
+        Returns:
+            Base or NoneType: an instance or none
+        """
+        try:
+            instance_config = self.store.get(name)
+        except FileNotFoundError:
```

Comment:
specifying the type of exception here is not the best solution IMHO, as we can have multiple stores, with every store might fail differently.

I think, either we don't specify any exception type (which is not the best solution) or we refactor all store backends to raise a specific exception type in case it failed to get the config for an instance.

Reply:
I made stores return None if instead doesn't exist instead of raising exceptions